ibis/backends/risingwave/__init__.py

```diff
@@ -0,0 +1,281 @@
+"""Risingwave backend."""
+
+from __future__ import annotations
+
+import inspect
+from typing import TYPE_CHECKING, Callable, Literal
+
+import sqlalchemy as sa
+
+import ibis.common.exceptions as exc
+import ibis.expr.operations as ops
+from ibis import util
+from ibis.backends.base.sql.alchemy import AlchemyCanCreateSchema, BaseAlchemyBackend
+from ibis.backends.risingwave.compiler import RisingwaveCompiler
+from ibis.backends.risingwave.datatypes import RisingwaveType
+from ibis.common.exceptions import InvalidDecoratorError
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    import ibis.expr.datatypes as dt
+
+
+def _verify_source_line(func_name: str, line: str):
+    if line.startswith("@"):
+        raise InvalidDecoratorError(func_name, line)
+    return line
+
+
+class Backend(BaseAlchemyBackend, AlchemyCanCreateSchema):
```

Comment:
It looks like a lot of code in this file are duplicated from the `postgres` backend. Would it make sense for the RisingWave backend to directly inherit from the `postgres` backend instead?

Reply:
Good suggestion! That make sense. Let me try.

Reply:
Since the postgres are going to move to sqlglot in the next future, let's first keep them separated as we have never examined sqlglot for risingwave. After the migration of Pg is done, we can then consider whether we can reuse/inherit pg's backend when we are migrating risingwave support from sqlalchemy to sqlglot. Does it make sense?